### ISSUE TITLE
Diffs (when all packages loaded) in tests under GAP stable-4.10 branch

### ISSUE THREAD

**Issue by olexandr-konovalov** [gap-packages/sgpviz#3]
When all packages are loaded, there are diffs in the tests: see e.g. https://travis-ci.org/gap-system/gap-docker-pkg-tests-stable-4.9/jobs/392042888

```
########> Diff in /home/gap/inst/gap-stable-4.9/pkg/SgpViz-0.999.1/tst/testall\
.tst:68
# Input is:
f1 := SemigroupFactorization(poi3,el1);
# Expected output:
[ [ Transformation( [ 1, 3, 4, 4 ] ), Transformation( [ 2, 4, 3, 4 ] ) ] ]
# But found:
WARNING: you are possibly using twice the same generator and this may cause pr\
\
oblems...
[ [ Transformation( [ 1, 3, 4, 4 ] ), Transformation( [ 4, 1, 2, 4 ] ) ] ]
########
########> Diff in /home/gap/inst/gap-stable-4.9/pkg/SgpViz-0.999.1/tst/testall\
.tst:70
# Input is:
f1[1][1] * f1[1][2] = el1;
# Expected output:
true
# But found:
false
########
########> Diff in /home/gap/inst/gap-stable-4.9/pkg/SgpViz-0.999.1/tst/testall\
.tst:72
# Input is:
SemigroupFactorization(poi3,[el1,el2]);
# Expected output:
[ [ Transformation( [ 1, 3, 4, 4 ] ), Transformation( [ 2, 4, 3, 4 ] ) ], 
  [ Transformation( [ 2, 4, 3, 4 ] ) ] ]
# But found:
WARNING: you are possibly using twice the same generator and this may cause pr\
\
oblems...
[ [ Transformation( [ 1, 3, 4, 4 ] ), Transformation( [ 4, 1, 2, 4 ] ) ], 
  [ Transformation( [ 2, 4, 3, 4 ] ), Transformation( [ 1, 3, 4, 4 ] ) ] ]
########
########> Diff in /home/gap/inst/gap-stable-4.9/pkg/SgpViz-0.999.1/tst/testall\
.tst:86
# Input is:
DisplayEggBoxOfDClass(D);
# Expected output:
[ [  1,  1,  0,  0 ],
  [  1,  1,  0,  0 ],
  [  0,  0,  1,  1 ],
  [  0,  0,  1,  1 ] ]
# But found:
[ [  1,  0,  1,  0 ],
  [  0,  1,  0,  1 ],
  [  0,  1,  0,  1 ],
  [  1,  0,  1,  0 ] ]
########
########> Diff in /home/gap/inst/gap-stable-4.9/pkg/SgpViz-0.999.1/tst/testall\
.tst:107
# Input is:
rcg := RightCayleyGraphAsAutomaton(b21);
# Expected output:
< deterministic automaton on 2 letters with 6 states >
# But found:
WARNING: you are possibly using twice the same generator and this may cause pr\
\
oblems...
< deterministic automaton on 2 letters with 6 states >
########
SgpViz package: testall.tst
msecs: 52
#I  Errors detected while testing package sgpviz 0.999.1
#I  using the test file `tst/testall.tst'
#I  RunPackageTests("sgpviz", "0.999.1", "tst/testall.tst", true): runtime 
12488
```
I have checked and this is triggered after loading the Semigroups package.

**Comment by olexandr-konovalov** [comment]
This is still the case with GAP stable-4.10 branch, see e.g. https://travis-ci.org/gap-system/gap-docker-pkg-tests-stable-4.10-staging/jobs/429206409

It will be good to fix this in time for GAP 4.10 release.

**Comment by ManuelAFDelgado** [comment]
The issue is fixed in Version 0.999.1. Thanks a lot, @alex-konovalov .

**Comment by olexandr-konovalov** [comment]
@ManuelAFDelgado Thank you! Those problems gone, one new appeared - please see https://travis-ci.org/gap-system/gap-docker-pkg-tests-stable-4.10-staging/jobs/435203238. It occurs only when all packages are loaded:

```
testing: /home/gap/inst/gap-stable-4.10/pkg/SgpViz-0.999.2/tst/testall.tst
########> Diff in /home/gap/inst/gap-stable-4.10/pkg/SgpViz-0.999.2/tst/testal\
l.tst:118
# Input is:
DotForDrawingDClassOfElement(poi3,Transformation([1,4,3,4]));
# Expected output:
"digraph  DClassOfElement {\ngraph [center=yes,ordering=out];\nnode [shape=pla\
\
intext];\nedge [color=cornflowerblue,arrowhead=none];\n1 [label=<\n<TABLE BORD\
\
ER=\"0\" CELLBORDER=\"0\" CELLPADDING=\"0\" CELLSPACING=\"0\" PORT=\"1\">\n<TR\
\
><TD BORDER=\"0\"><TABLE CELLSPACING=\"0\"><TR><TD BGCOLOR=\"white\" BORDER=\"\
\
0\">*abc</TD></TR>\n</TABLE></TD><TD BORDER=\"0\"><TABLE CELLSPACING=\"0\"><TR\
\
><TD BGCOLOR=\"white\" BORDER=\"0\">a</TD></TR>\n</TABLE></TD><TD BORDER=\"0\"\
\
><TABLE CELLSPACING=\"0\"><TR><TD BGCOLOR=\"white\" BORDER=\"0\">ab</TD></TR>\
\
\n</TABLE></TD></TR>\n<TR><TD BORDER=\"0\"><TABLE CELLSPACING=\"0\"><TR><TD BG\
\
COLOR=\"white\" BORDER=\"0\">bc</TD></TR>\n</TABLE></TD><TD BORDER=\"0\"><TABL\
\
E CELLSPACING=\"0\"><TR><TD BGCOLOR=\"white\" BORDER=\"0\">*bca</TD></TR>\n</T\
\
ABLE></TD><TD BORDER=\"0\"><TABLE CELLSPACING=\"0\"><TR><TD BGCOLOR=\"white\" \
\
BORDER=\"0\">b</TD></TR>\n</TABLE></TD></TR>\n<TR><TD BORDER=\"0\"><TABLE CELL\
\
SPACING=\"0\"><TR><TD BGCOLOR=\"white\" BORDER=\"0\">c</TD></TR>\n</TABLE></TD\
\
><TD BORDER=\"0\"><TABLE CELLSPACING=\"0\"><TR><TD BGCOLOR=\"white\" BORDER=\"\
\
0\">ca</TD></TR>\n</TABLE></TD><TD BORDER=\"0\"><TABLE CELLSPACING=\"0\"><TR><\
\
TD BGCOLOR=\"white\" BORDER=\"0\">*cab</TD></TR>\n</TABLE></TD></TR>\n</TABLE>\
\
>];\n}\n"
# But found:
"digraph  DClassOfElement {\ngraph [center=yes,ordering=out];\nnode [shape=pla\
\
intext];\nedge [color=cornflowerblue,arrowhead=none];\n1 [label=<\n<TABLE BORD\
\
ER=\"0\" CELLBORDER=\"0\" CELLPADDING=\"0\" CELLSPACING=\"0\" PORT=\"1\">\n<TR\
\
><TD BORDER=\"0\"><TABLE CELLSPACING=\"0\"><TR><TD BGCOLOR=\"white\" BORDER=\"\
\
0\">*abc</TD></TR>\n</TABLE></TD><TD BORDER=\"0\"><TABLE CELLSPACING=\"0\"><TR\
\
><TD BGCOLOR=\"white\" BORDER=\"0\">abca</TD></TR>\n</TABLE></TD><TD BORDER=\"\
\
0\"><TABLE CELLSPACING=\"0\"><TR><TD BGCOLOR=\"white\" BORDER=\"0\">abcab</TD>\
\
</TR>\n</TABLE></TD></TR>\n<TR><TD BORDER=\"0\"><TABLE CELLSPACING=\"0\"><TR><\
\
TD BGCOLOR=\"white\" BORDER=\"0\">bc</TD></TR>\n</TABLE></TD><TD BORDER=\"0\">\
\
<TABLE CELLSPACING=\"0\"><TR><TD BGCOLOR=\"white\" BORDER=\"0\">*bca</TD></TR>\
\
\n</TABLE></TD><TD BORDER=\"0\"><TABLE CELLSPACING=\"0\"><TR><TD BGCOLOR=\"whi\
\
te\" BORDER=\"0\">bcab</TD></TR>\n</TABLE></TD></TR>\n<TR><TD BORDER=\"0\"><TA\
\
BLE CELLSPACING=\"0\"><TR><TD BGCOLOR=\"white\" BORDER=\"0\">c</TD></TR>\n</TA\
\
BLE></TD><TD BORDER=\"0\"><TABLE CELLSPACING=\"0\"><TR><TD BGCOLOR=\"white\" B\
\
ORDER=\"0\">ca</TD></TR>\n</TABLE></TD><TD BORDER=\"0\"><TABLE CELLSPACING=\"0\
\
\"><TR><TD BGCOLOR=\"white\" BORDER=\"0\">*cab</TD></TR>\n</TABLE></TD></TR>\n\
\
</TABLE>>];\n}\n"
########
########> Diff in /home/gap/inst/gap-stable-4.10/pkg/SgpViz-0.999.2/tst/testal\
l.tst:136
# Input is:
Print(last);
# Expected output:
digraph  DClassOfElement {
graph [center=yes,ordering=out];
node [shape=plaintext];
edge [color=cornflowerblue,arrowhead=none];
1 [label=<
<TABLE BORDER="0" CELLBORDER="0" CELLPADDING="0" CELLSPACING="0" PORT="1">
<TR><TD BORDER="0"><TABLE CELLSPACING="0"><TR><TD BGCOLOR="white" BORDER="0">*\
\
abc</TD></TR>
</TABLE></TD><TD BORDER="0"><TABLE CELLSPACING="0"><TR><TD BGCOLOR="white" BOR\
\
DER="0">a</TD></TR>
</TABLE></TD><TD BORDER="0"><TABLE CELLSPACING="0"><TR><TD BGCOLOR="white" BOR\
\
DER="0">ab</TD></TR>
</TABLE></TD></TR>
<TR><TD BORDER="0"><TABLE CELLSPACING="0"><TR><TD BGCOLOR="white" BORDER="0">b\
\
c</TD></TR>
</TABLE></TD><TD BORDER="0"><TABLE CELLSPACING="0"><TR><TD BGCOLOR="white" BOR\
\
DER="0">*bca</TD></TR>
</TABLE></TD><TD BORDER="0"><TABLE CELLSPACING="0"><TR><TD BGCOLOR="white" BOR\
\
DER="0">b</TD></TR>
</TABLE></TD></TR>
<TR><TD BORDER="0"><TABLE CELLSPACING="0"><TR><TD BGCOLOR="white" BORDER="0">c\
\
</TD></TR>
</TABLE></TD><TD BORDER="0"><TABLE CELLSPACING="0"><TR><TD BGCOLOR="white" BOR\
\
DER="0">ca</TD></TR>
</TABLE></TD><TD BORDER="0"><TABLE CELLSPACING="0"><TR><TD BGCOLOR="white" BOR\
\
DER="0">*cab</TD></TR>
</TABLE></TD></TR>
</TABLE>>];
}
# But found:
digraph  DClassOfElement {
graph [center=yes,ordering=out];
node [shape=plaintext];
edge [color=cornflowerblue,arrowhead=none];
1 [label=<
<TABLE BORDER="0" CELLBORDER="0" CELLPADDING="0" CELLSPACING="0" PORT="1">
<TR><TD BORDER="0"><TABLE CELLSPACING="0"><TR><TD BGCOLOR="white" BORDER="0">*\
\
abc</TD></TR>
</TABLE></TD><TD BORDER="0"><TABLE CELLSPACING="0"><TR><TD BGCOLOR="white" BOR\
\
DER="0">abca</TD></TR>
</TABLE></TD><TD BORDER="0"><TABLE CELLSPACING="0"><TR><TD BGCOLOR="white" BOR\
\
DER="0">abcab</TD></TR>
</TABLE></TD></TR>
<TR><TD BORDER="0"><TABLE CELLSPACING="0"><TR><TD BGCOLOR="white" BORDER="0">b\
\
c</TD></TR>
</TABLE></TD><TD BORDER="0"><TABLE CELLSPACING="0"><TR><TD BGCOLOR="white" BOR\
\
DER="0">*bca</TD></TR>
</TABLE></TD><TD BORDER="0"><TABLE CELLSPACING="0"><TR><TD BGCOLOR="white" BOR\
\
DER="0">bcab</TD></TR>
</TABLE></TD></TR>
<TR><TD BORDER="0"><TABLE CELLSPACING="0"><TR><TD BGCOLOR="white" BORDER="0">c\
\
</TD></TR>
</TABLE></TD><TD BORDER="0"><TABLE CELLSPACING="0"><TR><TD BGCOLOR="white" BOR\
\
DER="0">ca</TD></TR>
</TABLE></TD><TD BORDER="0"><TABLE CELLSPACING="0"><TR><TD BGCOLOR="white" BOR\
\
DER="0">*cab</TD></TR>
</TABLE></TD></TR>
</TABLE>>];
}
########
     540 ms (336 ms GC) and 144MB allocated for testall.tst
-----------------------------------
total       540 ms (336 ms GC) and 144MB allocated
#I  Errors detected while testing
```

**Comment by ManuelAFDelgado** [comment]
Thank you @alex-konovalov for pointing out this new issue. 
It is caused by the use of different factorization algorithms... Instead of checking the equality of the strings obtained I will just make some particular tests, such as testing the equality of the number of nodes, the number of idempotents, etc.
Please let me know if you think that I should immediately produce a new version. As the problem is harmless, I would delay it a couple of weeks, until getting a little bit familiar with travis and codecov (which most probably will lead me to do a few more changes).

**Comment by olexandr-konovalov** [comment]
Thanks @ManuelAFDelgado - couple of weeks sounds good. The plan to modify tests looks plausible, but do have some tests which check the equality of strings too, perhaps for some simple examples which look always the same.

**Comment by olexandr-konovalov** [comment]
Is https://github.com/gap-packages/sgpviz/releases/tag/v0.999.4 fixing this?

**Comment by ManuelAFDelgado** [comment]
I managed to made a new release of sgpviz (0.999.4) which hopefully solves the problem. Sorry for the delay.

**Comment by olexandr-konovalov** [comment]
Thanks, it's just in time - I was about to freeze updates, when just discovered it.

**Comment by olexandr-konovalov** [comment]
0.999.4 is included in GAP 4.10.0 and runs well - thanks!
